jax/_src/nn/functions.py

```diff
@@ -25,10 +26,12 @@
 from jax.scipy.special import logsumexp as _logsumexp
 import jax.numpy as jnp
 
+Array = Any
+
 # activations
 
 @custom_jvp
-def relu(x):
+def relu(x: Array) -> Array:
```

Comment:
No action required, but I suspect that ultimately we will need to accept an `ArrayLike` type (that includes a number of non-arrays, e.g., scalars,) but return an `Array`. Obviously this doesn't matter yet given `Array = Any`.

Reply:
Yeah, I'm looking to see what the JAX team comes up with.  NumPy 1.20 already has type annotations for `ndarray`, and it looks like they're making it generic on shape and dtype soon.